org/postgresql/jdbc2/TypeInfoCache.java

```diff
@@ -150,9 +153,14 @@ public synchronized void addCoreType(String pgTypeName, Integer oid, Integer sql
         Character delim = ',';
         _arrayOidToDelimiter.put(oid, delim);
 
-        String pgArrayTypeName = "_" + pgTypeName;
```

Comment:
Sorry, I tried adding `_oidToPgName.put(arrayOid, pgArrayTypeName);` and it has not make second test passing.
I added this second test to #592 PR.

> Let me know if you want me to finish it up. 
I don't really get all the consequences of these changes. So I would prefer to take tiny QA-role here. Fell free to use tests from #592 in any way.

>And what have you decided as array format.
Sorry I am not the decision making person here. I only say that I personally can live with breaking compatibility change.